File: Solutions/797. All Paths From Source to Target/797. All Paths From Source to Target.java
class Solution {
    public List<List<Integer>> allPathsSourceTarget(int[][] g) {
        List<List<Integer>> res=new ArrayList<>();
        List<Integer> path=new ArrayList<>();
        path.add(0);
        dfs(g,res,path,0);
        return res;
    }
    private void dfs(int[][] g,List<List<Integer>> res,List<Integer> path,int node){
        if(node==g.length-1)
            res.add(new ArrayList<Integer>(path));
        for(var i:g[node]){
            path.add(i);
            dfs(g,res,path,i);
            path.remove(path.size()-1);
        }
    }
}

File: Solutions/797. All Paths From Source to Target/797. All Paths From Source to Target.py
class Solution:
    def allPathsSourceTarget(self, g: List[List[int]]) -> List[List[int]]:
        def dfs(g,res,path,node):
            if node==len(g)-1:
                res.append(path[:])
            for i in g[node]:
                path.append(i)
                dfs(g,res,path,i)
                path.pop(-1)
        res,path=[],[]
        path.append(0)
        dfs(g,res,path,0)
        return res